pkg/trie/database.go

```diff
@@ -39,10 +38,36 @@ func (t *Trie) Load(db db.DBGetter, rootHash common.Hash) error {
 		return fmt.Errorf("cannot decode root node: %w", err)
 	}
 
+	err = loadStorageValue(db, root)
+	if err != nil {
+		return fmt.Errorf("while decoding storage value for root: %w", err)
+	}
+
 	t.root = root
 	t.root.MerkleValue = rootHashBytes
 
-	return t.loadNode(db, t.root)
+	err = t.loadNode(db, t.root)
+	if err != nil {
+		return err
+	}
+
+	for _, key := range t.GetKeysWithPrefix(ChildStorageKeyPrefix) {
```

Comment:
Weren't we loading child tries? 😱

Reply:
I just moved from loadNode() and bring to Load() method, since we don't need to load a child trie every time we load a node in the trie